cmd/kube-controller-manager/app/options/daemonsetcontroller.go

```diff
@@ -52,5 +56,8 @@ func (o *DaemonSetControllerOptions) Validate() []error {
 	}
 
 	errs := []error{}
+	if o.ConcurrentDaemonSetSyncs < 1 {
+		errs = append(errs, fmt.Errorf("concurrent-daemonset-syncs must be greater than 0, but got %d", o.ConcurrentDaemonSetSyncs))
```

Comment:
will this break existing users that are not setting the flag?

Reply:
I see, so this is defaulted here

https://github.com/kubernetes/kubernetes/blob/d001d5684e69b08e120dc328917925318c5f324c/pkg/controller/daemon/config/v1alpha1/defaults.go#L32-L36

I always forget how the flow goes, config override flags?

Reply:
IIUC because we have defaulting to 2 [here](https://github.com/kubernetes/kubernetes/blob/daef8c2419a638d3925e146d0f5a6b217ea69b74/pkg/controller/daemon/config/v1alpha1/defaults.go#L23-L36) that we use in the definition of the `"concurrent-daemonset-syncs"` flag above (third argument of the `Int32Var` call - agreed it's not terribly explicit...), that field would be set to 2 in that place if a user does not use this flag.

Last year we did the exact same thing for the Job controller without any problems: https://github.com/kubernetes/kubernetes/pull/117138.

Reply:
ok, I tested that omitting the flag keep the default to 2 and setting explicit to 0 fails

Reply:
just in case you want to add more coverage
```diff
diff --git a/cmd/kube-controller-manager/app/options/options_test.go b/cmd/kube-controller-manager/app/options/options_test.go
index e32e3bd58c8..6c132be90d3 100644
--- a/cmd/kube-controller-manager/app/options/options_test.go
+++ b/cmd/kube-controller-manager/app/options/options_test.go
@@ -515,6 +515,13 @@ func TestValidateFlags(t *testing.T) {
                        },
                        wantErr: true,
                },
+               {
+                       name: "Invalid concurrent daemonset controller",
+                       flags: []string{
+                               "--concurrent-daemonset-syncs=0",
+                       },
+                       wantErr: true,
+               },
        }
```

Reply:
Thx for double-checking Antonio :)